### ISSUE TITLE
Can training results be adjusted?

### ISSUE THREAD

**Issue by D-Mad** [facebookresearch/vsait#5]
Great project!,I have some questions that I'm wondering.
- Normally, if you run gta2cityscape, how much is the max_step setting?
_ What can be adjusted? or adjust what parameters if i want the GTA image transition to be a bit more cityscape-like, is that possible? Can I adjust the level of the conversion? For example, after the training result, I feel unsatisfied for example

**Comment by jdtheiss** [comment]
Hi, thanks for your interest in our work! There are a few different parameters you could consider changing:

- In the paper, we used `max_steps = 400000`, which should improve the results compared to the default setting of 200k. 
- Another simple parameter to tune would be the [VSA_LOSS_COEF](https://github.com/facebookresearch/vsait/blob/c0abe5d1fce2cf965bb8af96ecdc611caeef0189/configs/vsait.yaml#L22) in the model config. A lower value will give greater relative weight to the GAN loss.
- You can also try changing the [LSH](https://github.com/facebookresearch/vsait/blob/main/configs/vsait.yaml#L16) node in the model config to use e.g. dilation when encoding extracted features into the VSA hyperspace. Each layer of extracted features is encoded using a convolutional filter, so you can use the same keyword arguments as you would for `nn.Conv2d`. 
- Furthermore, you can try changing the `RETURN_NODES` or use a different pre-trained model for the [feature extractor](https://github.com/facebookresearch/vsait/blob/main/configs/vsait.yaml#L8), which can be set to any valid `torchvision.models.*` network. Note that the number of `RETURN_NODES` should match the number of kernel sizes/strides/etc. in the `LSH` node of the model config.
- Finally, you can also change the mapping network by defining `MAP_NET_KWARGS` in the `TASK_MODEL` node of the model config. The [default](https://github.com/facebookresearch/vsait/blob/main/models/iit.py#L122) for the mapping network is a 2-layer convolutional network with kernel_size 1 and padding 0 (note that the input and output tensor sizes must match). It may be useful to look at the [MLP](https://github.com/facebookresearch/vsait/blob/c0abe5d1fce2cf965bb8af96ecdc611caeef0189/models/networks.py#L132) and [ConvBlock](https://github.com/facebookresearch/vsait/blob/c0abe5d1fce2cf965bb8af96ecdc611caeef0189/models/networks.py#L26) classes as well for other arguments. Here's an example for defining the mapping network in the model config:
```
MAP_NET_KWARGS:
    kernel_size: 3
    padding: 1
```

**Comment by D-Mad** [comment]
Thank you very much for the very quick reply. Thank you once again.

I trained over 400k but the loss is only 1.02, should I train more, what should the loss be to look the best?,I used 20k images gta and 2k cityscapes
The default batch size is 1 , should I increase it ? I'm training on rtx4090

**Comment by jdtheiss** [comment]
Unfortunately, the total loss is likely not very informative, but monitoring each individual loss should be useful. The "discr"/"adv" losses should stabilize over time whereas the "vsa" loss should decrease over time. Increasing the batch size may help as well.

**Comment by jdtheiss** [comment]
Hopefully the comments above were helpful, but feel free to reopen the issue if you have more questions related to training options.
